management-api-server/src/test/resources/com/datastax/mgmtapi/operator-sample.json

```diff
@@ -0,0 +1,78 @@
+{
```

Comment:
Good question. There was already a test that was calling `configureNode` that passed yaml [here](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/test/java/com/datastax/mgmtapi/LifecycleIT.java#L141-L159). However, the API actually takes JSON or YAML as the request body (JSON [here](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/java/com/datastax/mgmtapi/resources/LifecycleResources.java#L325-L342), YAML [here](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/java/com/datastax/mgmtapi/resources/LifecycleResources.java#L344-L358)). If you look, you'll see the method that handles JSON essentially tries to read the Request bosy as a JSON string, then convert that JSON into YAML, then just call the `configureNode` method that handles YAML and does the real work.

Having 2 methods both handling the same API endpoint (POST to `configure` effectively) causes the OpenAPI spec generation to fail (it actually only documents one of the methods, so the doc only shows JSON or only shows YAML as the Request body type, I can't remember which). That is what led to the wonderful hack of marking both those methods as `@Hidden` and manually documenting them [here](https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/resources/openapi-configuration.json).

In this PR, I have fixed all of that some. First, I consolidated the OpenAPI annotations for `@Consumes` into a single method that accepts YAML or JSON now:
https://github.com/k8ssandra/management-api-for-apache-cassandra/pull/325/files#diff-f0ea4ea1635e425b16c7279f1a2e1c3f5eec7fd96cb5e6c2c14774af2533519bR315
I changed the implementation of that method a little since it can't assume it's only handling JSON requests. So now it tries to parse the request body into JSON, then tries to convert that to YAML.
https://github.com/k8ssandra/management-api-for-apache-cassandra/pull/325/files#diff-f0ea4ea1635e425b16c7279f1a2e1c3f5eec7fd96cb5e6c2c14774af2533519bR364-R368
If that fails, it might be that the request body was already yaml, so in the catch block, it just passes the request body as a string to the `configureNode` method that deals with YAML:
https://github.com/k8ssandra/management-api-for-apache-cassandra/pull/325/files#diff-f0ea4ea1635e425b16c7279f1a2e1c3f5eec7fd96cb5e6c2c14774af2533519bR369-R371

Looking at it now, I kept the original method names, but I should probably rename them as they don't do what the names indicate.

Reply:
Totally forgot to mention, the yaml version of this file already existed in the test resources. I created this json version from that file simply to test that the json endpoint works as well as the yaml endpoint.

Reply:
OK, this isn't a bad way to do it, however could we possibly control the expected type using the `Content-Type` header instead of using a try-catch block? That seems like the principled way to do it. Does that make things harder for you with the spec generation, or easier?

Reply:
The `@Consumes` annotation effectively defines the `Content-Type`s that are allowed. For the `/configure` endpoint, it is `application/json`, `application/yaml` and `text/yaml`.

Prior to this change, there were 2 methods attached to the `/configure` endpoint; one that took JSON, and one that took YAML.
https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/java/com/datastax/mgmtapi/resources/LifecycleResources.java#L326
```java
  @POST
  @Consumes("application/json")
  @Operation(
      description = "Configure Cassandra. Will fail if Cassandra is already started",
      operationId = "configureNodeJson")
  @Produces(MediaType.TEXT_PLAIN)
  public synchronized Response configureNodeJson(
      @QueryParam("profile") String profile, String config) {
```

https://github.com/k8ssandra/management-api-for-apache-cassandra/blob/master/management-api-server/src/main/java/com/datastax/mgmtapi/resources/LifecycleResources.java#L348

```java
  @POST
  @Consumes({"application/yaml", "text/yaml"})
  @Operation(
      description = "Configure Cassandra/DSE. Will fail if Cassandra/DSE is already started",
      operationId = "configureNode")
  @Produces(MediaType.TEXT_PLAIN)
  public synchronized Response configureNode(@QueryParam("profile") String profile, String yaml) {
```
However, the OpenAPI document generation did not like 2 methods associated with the same (POST) operation on the same path, with  the Content-Type of the request being the only difference. Initially, this led me to excluding both methods from the auto-generated OpenAPI doc and maintaining the definition it should look like in that static file (the one that gets merged with the auto-generated stuff to produce the final OpenAPI document).

This PR tries to reduce the manual maintenance required to keep the doc accurate. To do so, I have to consolidate the POST to `/configure` into a single method so the auto-generation doesn't choke, which means one method that accepts both Content-Types.
```java
  public synchronized Response configureNodeJson(
      @QueryParam("profile") String profile,
      @RequestBody(
              content = {
                @Content(mediaType = "application/yaml"),
                @Content(mediaType = "text/yaml"),
                @Content(mediaType = "application/json")
              },
              required = true,
              description = "JSON or YAML configuration overrides")
          String config) {
```
Unfortunately, doing that means I can no longer tell the request body format as now I have a String representation of either JSON or YAML configuration. If you look at the old implementation, the method that handled JSON requests just converted the JSON to YAML and then called the method that handled YAML requests, which did all the work. The benefit however is that if parsing the request body as JSON and then converting to YAML failed, we could return an error saying "you have bad JSON".

With my changes, bad JSON will just get attempted again as YAML, and likely still be bad, which will return an error indicating a bad request. It's not as specific, but the behavior for the success paths is exactly the same.

There is likely a better way of doing this, but I don't want to change the `v0/configure` endpoint and how it currently works. We can address it in v2 (maybe we just expose 2 endpoints to avoid the doc generation and mixing requestBody Content-Type handling into a single method) if we want, but it's out of scope for the tickets in this phase.

Lastly, if this change, which arguably is out of scope as well, is too much for this PR, I can revert it, but it will bring back the hand-managed definition for the `confugre` endpoint.

Reply:
Per discussions with @emerkle826 we are deferring this work until later. We will stick with two endpoints for now and manually generate the OpenAPI doc.